[Swift in WebKit] Compile GroupActivities code on the Public SDK

rr-codes opened the pull request:
#### 94c77bbbd1e95acc18da009b43f1f82ab0b64b91
<pre>
[Swift in WebKit] Compile GroupActivities code on the Public SDK
<a href="https://bugs.webkit.org/show_bug.cgi?id=293473">https://bugs.webkit.org/show_bug.cgi?id=293473</a>
<a href="https://rdar.apple.com/151902343">rdar://151902343</a>

Reviewed by Andy Estes and Elliott Williams.

Make it possible for GroupActivities-related code to be compiled on the public SDK.

* Source/WebKit/Configurations/WebKit.xcconfig:

Like WKS, add include paths on the public SDK so that it will use the handwritten SPI swiftinterfaces.

* Source/WebKit/Platform/spi/Cocoa/GroupActivities_SPI.swiftinterface: Added.

Create the .swiftinterface for GroupActivities with the SPI symbols that are used.

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:

Remove the internal SDK guards.

* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:

Import the dummy SPI module when using the public SDK.

Canonical link: <a href="https://commits.webkit.org/295407@main">https://commits.webkit.org/295407@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b506f0ae6ec6d322e8990d7aa0bb96c957b9dbc0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104993 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24707 "Hash b506f0ae for PR 45815 does not build (failure)") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15129 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110210 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55672 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25115 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33251 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79725 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107999 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19545 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94758 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60032 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19289 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55050 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89011 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12882 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112707 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32158 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23662 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88806 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32523 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90984 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88434 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22542 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33328 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11111 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/27505 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32082 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37454 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31875 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35216 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33434 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -65,7 +65,9 @@ BUILD_LIBRARY_FOR_DISTRIBUTION = YES;
-SWIFT_INCLUDE_PATHS = $(SRCROOT)/Modules/Internal;
+// Use handwritten SPI modules on public SDKs.
+SWIFT_INCLUDE_PATHS = $(SRCROOT)/Modules/Internal $(SWIFT_INCLUDE_PATHS_$(USE_INTERNAL_SDK));
+SWIFT_INCLUDE_PATHS_ = $(SRCROOT)/Platform/spi/visionos $(SRCROOT)/Platform/spi/ios  $(SRCROOT)/Platform/spi/Cocoa $(SRCROOT)/Platform/spi/Cocoa/Modules $(inherited);
```
```suggestion
SWIFT_INCLUDE_PATHS_ = $(SRCROOT)/Platform/spi/visionos $(SRCROOT)/Platform/spi/ios $(SRCROOT)/Platform/spi/Cocoa $(SRCROOT)/Platform/spi/Cocoa/Modules $(inherited);
```

aestes commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -7997,7 +7998,7 @@
 		CD8252E025D4918400862FD8 /* RemoteRemoteCommandListenerMessageReceiver.cpp */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.cpp.cpp; path = RemoteRemoteCommandListenerMessageReceiver.cpp; sourceTree = "<group>"; };
 		CD8252E125D4918500862FD8 /* RemoteRemoteCommandListenerMessages.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = RemoteRemoteCommandListenerMessages.h; sourceTree = "<group>"; };
-		CD95493526159004008372D9 /* libWebKitSwift.dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = libWebKitSwift.dylib; sourceTree = BUILT_PRODUCTS_DIR; };
+		CD95493526159004008372D9 /* .dylib */ = {isa = PBXFileReference; explicitFileType = "compiled.mach-o.dylib"; includeInIndex = 0; path = .dylib; sourceTree = BUILT_PRODUCTS_DIR; };
```
What happened here?

emw-apple replied:
I've seen this on my system too, but I don't know what's causing it yet. Usually happens because some build setting is conditionalized.

emw-apple commented on Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:
```diff
@@ -21,13 +21,21 @@
 @_spi(Safari) import GroupActivities
+#else
+import GroupActivities
+#if compiler(>=6.0)
+internal import GroupActivities_SPI
+#else
+@_implementationOnly import GroupActivities_SPI
+#endif
```
Do we actually care about making it an `internal import`? Since the open source build does not install or ship anywhere, it doesn't seems like it matters, but maybe it breaks the build in other ways?

rr-codes replied:
it does break the build if I remove it:

```
error: Unable to find module dependency: 'GroupActivities_SPI'
note: a dependency of Swift module 'WebKit': '/Users/richardrobinson/safari/OpenSource/WebKitBuild/Debug/WebKit.framework/Modules/WebKit.swiftmodule/arm64-apple-macos.swiftmodule'
note: a dependency of main module 'libWebKitSwift'
error: Unable to find module dependency: 'GroupActivities_SPI'
note: a dependency of Swift module 'WebKit': '/Users/richardrobinson/safari/OpenSource/WebKitBuild/Debug/WebKit.framework/Modules/WebKit.swiftmodule/arm64-apple-macos.swiftmodule'
note: a dependency of main module 'TestWebKitAPIBundle'
```

emw-apple reviewed (approved):
Restarted the ios-sim EWS job for you, I think you got unlucky and hit a flaky build failure. r=me once you undo that change the Xcode project.

rr-codes committed:
[Swift in WebKit] Compile GroupActivities code on the Public SDK
https://bugs.webkit.org/show_bug.cgi?id=293473
rdar://151902343

Reviewed by Andy Estes and Elliott Williams.

Make it possible for GroupActivities-related code to be compiled on the public SDK.

* Source/WebKit/Configurations/WebKit.xcconfig:

Like WKS, add include paths on the public SDK so that it will use the handwritten SPI swiftinterfaces.

* Source/WebKit/Platform/spi/Cocoa/GroupActivities_SPI.swiftinterface: Added.

Create the .swiftinterface for GroupActivities with the SPI symbols that are used.

* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesCoordinator.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSession.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.h:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/GroupActivitiesSessionNotifier.mm:
* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:

Remove the internal SDK guards.

* Source/WebKit/UIProcess/Cocoa/GroupActivities/WKGroupSession.swift:

Import the dummy SPI module when using the public SDK.

Canonical link: https://commits.webkit.org/295407@main
